### PR TITLE
Test Suite Update

### PR DESCRIPTION
**Pull request by lfields90**

### CODE REVIEW

File: app/controllers/reviews_controller.rb [hunk 1]
```diff
@@ -19,6 +19,27 @@ def create
...
+  def edit
+    @spot = Spot.find params[:spot_id]
+    @review = Review.find(params[:id])
+    @ratings = [1, 2, 3, 4, 5]
+  end
+
+  def update
+    @spot = Spot.find params[:spot_id]
+    @review = Review.find(params[:id])
+    @review.update(review_params)
```
Comment (cmkoller) [inline app/controllers/reviews_controller.rb]: There should be an `if` statement that checks to see if the review was successfully updated or not, and flashes a relevant error message

File: app/views/reviews/edit.html.erb [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  <%= f.label(:rating, 4, value: 4) %>
+
+  <%= f.radio_button(:rating, 5) %>
+  <%= f.label(:rating, 5, value: 5) %>
+
+  <%= f.label :body %>
+  <%= f.text_area :body %>
+
+  <%= f.submit "Edit Review" %>
+<% end %>
```
Comment (cmkoller) [inline app/views/reviews/edit.html.erb]: Indentation

File: spec/features/user_edits_a_review_spec.rb [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    review = FactoryGirl.create(:review, user_id: user.id, spot_id: spot.id)
+    visit edit_spot_review_path(spot, review)
+
+    choose '3'
+    fill_in 'Body', with: 'Pretty cool'
+    click_button("Edit review")
+
+    expect(page).to have_content(spot.name)
+    expect(page).to have_content(review.body)
+  end
```
Comment (cmkoller) [inline spec/features/user_edits_a_review_spec.rb:34]: There should be a test in here checking for error messages if the review is unsuccessfully edited

File: spec/models/spot_spec.rb [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+  end
+  it "is invalid without city" do
+    FactoryGirl.build(:spot, city: nil).should_not be_valid
+  end
+  it "is invalid without state" do
+    FactoryGirl.build(:spot, state: nil).should_not be_valid
+  end
+  it "is invalid without zip code" do
+    FactoryGirl.build(:spot, zip_code: nil).should_not be_valid
+  end
```
Comment (cmkoller) [inline spec/models/spot_spec.rb]: So, this approach works, but I would look into [shoulda-matchers](https://horizon.launchacademy.com/questions/315) as an easier and quicker way of testing your validations! (Which is really what you're doing here - checking that your model validates for different things)

File: spec/support/factory_girl.rb [hunk 5]
```diff
@@ -1,10 +1,31 @@
...
-    sequence(:email) {|n| "user#{n}@example.com" }
-    password 'password'
-    password_confirmation 'password'
+  factory :user do |f|
+    f.email { Faker::Internet.email }
+    f.password { Faker::Internet.password(10, 20) }
   end
+end
 
+FactoryGirl.define do
```
Comment (cmkoller) [inline spec/support/factory_girl.rb]: You only need `FactoryGirl.define do` once - _all_ your factories can go inside that one block you defined at the top:

```
FactoryGirl.define do
  factory :user do
    ...
  end

  factory :spot do 
    ...
  end
end
```

**Comment by cmkoller** [comment]
This PR should have a more descriptive name than "Model tests", since a lot of the work you're doing is unrelated to that!

File: spec/features/user_deletes_a_spot_spec.rb [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+
+feature "User deletes a spot" do
+
+  # As a user I want to be able to edit a spot I created
+  # so that the information is up-to-date.
+  # Acceptance criteria:
+  # User can edit the spot they created.
+
+  scenario 'I want to delete a study spot' do
+    skip
```
Comment (cmkoller) [inline spec/features/user_deletes_a_spot_spec.rb]: What's the `skip` in here for?

File: spec/features/user_adds_a_review_spec.rb [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+    user = FactoryGirl.create(:user)
+    spot = FactoryGirl.create(:spot)
+
+    visit new_user_session_path
+
+    fill_in 'Email', with: user.email
+    fill_in 'Password', with: user.password
+    click_button 'Log in'
+
+    visit spot_path(spot)
```
Comment (cmkoller) [inline spec/features/user_adds_a_review_spec.rb:14]: Sorry for not noticing this before, but lines 5-14 should be extracted into a `before(:each)` action so they're not duplicated in each test

File: spec/features/user_adds_a_review_spec.rb [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+    choose "5"
+    fill_in "Body", with: "Great study spot"
+
+    click_button "Add Review"
+
+    expect(page).to have_content("Review successfully added")
+    expect(page).to have_content("Great study spot")
+  end
+
+  feature "adding review" do
```
Comment (cmkoller) [inline spec/features/user_adds_a_review_spec.rb]: Why is there another `feature` line here? That shouldn't be here!

File: spec/features/user_adds_a_review_spec.rb [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+
+      fill_in "Body", with: "Great study spot"
+
+      click_button "Add Review"
+
+      expect(page).to have_content("Rating can't be blank")
+    end
+  end
+
+  feature "adding review" do
```
Comment (cmkoller) [inline spec/features/user_adds_a_review_spec.rb]: Too many `feature`s! :D

File: spec/features/user_adds_a_study_spot_spec.rb [hunk 10]
```diff
@@ -22,4 +22,16 @@
 
     expect(page).to have_content("Spot added.")
   end
+
+  feature "adding spot" do
```
Comment (cmkoller) [inline spec/features/user_adds_a_study_spot_spec.rb]: Get rid of this feature line - it should all be nested under the first `feature`

File: spec/features/user_reads_and_votes_on_reviews_spec.rb [hunk 11]
```diff
@@ -9,14 +9,17 @@
   # - User can see all reviews for a spot.
 
   pending scenario 'I want to see reviews for a spot' do
```
Comment (cmkoller) [inline spec/features/user_reads_and_votes_on_reviews_spec.rb]: This should probably not be pending any more...

File: spec/models/review_spec.rb [hunk 12]
```diff
@@ -0,0 +1,34 @@
...
+# =>                  spot_id: spot.id, body: nil).should_not be_valid
+# end
+# it "is invalid without spot id" do
+#   FactoryGirl.build(:review, user_id: user.id,
+#                     spot_id: nil).should_not be_valid
+# end
+# it "is invalid without user id" do
+#   FactoryGirl.build(:review, spot_id: spot.id,
+#                     user_id: nil).should_not be_valid
+# end
```
Comment (cmkoller) [inline spec/models/review_spec.rb]: Delete these comments! You can always go back and find it on Github if you need to!

File: spec/models/spot_spec.rb [hunk 13]
```diff
@@ -0,0 +1,38 @@
...
+# end
+# it "is invalid without city" do
+#   FactoryGirl.build(:spot, city: nil).should_not be_valid
+# end
+# it "is invalid without state" do
+#   FactoryGirl.build(:spot, state: nil).should_not be_valid
+# end
+# it "is invalid without zip code" do
+#   FactoryGirl.build(:spot, zip_code: nil).should_not be_valid
+# end
```
Comment (cmkoller) [inline spec/models/spot_spec.rb]: Delete these too!

File: spec/features/user_deletes_a_spot_spec.rb (hunk 6 above)
Comment (lfields90) [inline spec/features/user_deletes_a_spot_spec.rb]: It is so that I can skip this test. We haven't full incorporated the delete function yet.

Comment (cmkoller) [inline spec/features/user_deletes_a_spot_spec.rb]: It looks like you have a `destroy` action in your spots controller - you should either fully implement and test that feature, or not include it all! It's a bad idea to merge half-finished work into master.

**Pull request merged**